src/yarp.c

```diff
@@ -1594,10 +1594,19 @@ void
 parse_arguments(yp_parser_t *parser, yp_node_t *arguments) {
   if (!accept(parser, YP_TOKEN_PARENTHESIS_RIGHT)) {
     while (parser->current.type != YP_TOKEN_EOF) {
-      yp_node_t *expression = parse_expression(parser, BINDING_POWER_NONE, "Expected to be able to parse an argument.");
-      if (expression->type == YP_NODE_MISSING_NODE) break;
+      yp_node_t *argument;
+
+      if (accept(parser, YP_TOKEN_DOT_DOT_DOT)) {
+	if (!yp_token_list_includes(&parser->current_scope->as.scope.locals, &parser->previous)) {
+	  expect(parser, YP_TOKEN_DOT_DOT_DOT, "unexpected ... when parent method is not forwarding.");
```

Comment:
This should add the error directly and not go through the `expect` method. Otherwise it could end up with a weird case where it accepts `foo(... ...)`. You can use `yp_error_list_append`